plugins/leemons-plugin-academic-calendar/frontend/src/pages/private/regional/index.js

```diff
@@ -120,10 +119,14 @@ export default function RegionalCalendars() {
     scroll: 0,
   });
 
+  const { data: regionalConfigs, refetch: refetchConfigs } = useListRegionalConfigs(
+    store?.center?.id,
+    { enabled: !!store?.center?.id }
+  );
+
   async function loadRegionalConfigs() {
-    const { regionalConfigs } = await listRegionalConfigsRequest(store?.center?.id);
-    store.regionalConfigs = regionalConfigs;
-    setRegConfigs(regionalConfigs);
+    refetchConfigs();
+    store.regionalConfigs = regionalConfigs?.regionalConfigs;
```

Comment:
Does not work as expected, refetch function will trigger a refetch effectively, but it won't update regionalConfigs var inside of this context.

In other words, store.regionalConfigs will potentialy have a missmatching version of the regionalConfigs.

Reply:
I've made a refactor on entire archive. Add React Query hooks to all queries and mutations. Remove store and renders.